cmd/carapace/cmd/lazyinit/nushell.go

```diff
@@ -5,15 +5,27 @@ import "fmt"
 func Nushell(completers []string) string {
 	snippet := `%v%v
 
-let carapace_completer = {|spans| 
-  carapace $spans.0 nushell $spans | from json
+let carapace_completer = {|spans|
+  # if the current command is an alias, get it's expansion
+  let expanded_alias = (scope aliases | where name == $spans.0 | get -i 0 | get -i expansion)
+
+  # overwrite
+  let spans = (if $expanded_alias != null  {
+    # put the first word of the expanded alias first in the span
+    $spans | skip 1 | prepend ($expanded_alias | split row " ")
```

Comment:
Is the expanded alias provided by nushell a single string?
Splitting by space works in most cases but arguments can contain a space as well (e.g. `gh issue list --label 'help wanted'`).

I'm also not getting positional completion for `alias gc = git checkout` (flags work).
If the expanded alias is `git checkout ` (note the space suffix) the third *empty* span might be missing.

Reply:
No, wait - the empty span should be present in the original spans :thinking: .

Reply:
Nothing much we can do if it is a string (it's just a temporary workaround anyway), but would be good to have positional completion working.

Reply:
Seems carapace gets `checkout` passed twice:

```go
2024/01/02 15:52:03.164124 nushell []string{"carapace", "_carapace", "nushell", "git", "checkout", "checkout", ""}
```

Reply:
I mean, if carapace does tokenization on its own, maybe we can pass the non splitted string?

Reply:
It has [tokenization](https://github.com/rsteube/carapace-shlex) and it is used for `bash`.
But that is more a workaround for wordbreaks and redirects being passed to the completion function.
Writing a fully working lexer for all shells is a major task and not desirable.

I'm fine with the current state, I'm more concered about the other issue.
Don't get why `checkout` gets passed twice, the prepending works when i try it manually in my shell.

Reply:
Is it my setup? Does the following work for you?:

```nushell
alias gc = git checkout
gc <TAB>
```

Reply:
same behavior.. many direct non flag completes get ignored. :( Same for

```nix
          gad = "git add";
          gbr = "git branch";
```

git merge is working.. who knows why :D

```nix
          gmg = "git merge";
```

Reply:
I think I got it. Nushell **does** indeed expand the alias, but just does not replace the first span :see_no_evil: .
```nushell
alias gc = git checkout
gc <TAB>
# original spans = [gc checkout]
```

Reply:
Seems it is what you wanted to do but missed adding `first`.

```diff
diff --git a/cmd/carapace/cmd/lazyinit/nushell.go b/cmd/carapace/cmd/lazyinit/nushell.go
index a1f4025f..6d59bd48 100644
--- a/cmd/carapace/cmd/lazyinit/nushell.go
+++ b/cmd/carapace/cmd/lazyinit/nushell.go
@@ -12,7 +12,7 @@ let carapace_completer = {|spans|
   # overwrite
   let spans = (if $expanded_alias != null  {
     # put the first word of the expanded alias first in the span
-    $spans | skip 1 | prepend ($expanded_alias | split row " ")
+    $spans | skip 1 | prepend ($expanded_alias | split row " " | first)
   } else {
     $spans
   })
```

Reply:
Worked! Take 1 replaces now only the alias (first span) and it just worked :D